docs/src/components/header/HeaderNavBar.tsx

```diff
@@ -254,25 +224,24 @@ export default function HeaderNavBar() {
 
   return (
     <Navigation>
-      <ul ref={navRef} role="menubar" onKeyDown={handleLeftRightArrow}>
+      <ul ref={navRef} onKeyDown={handleKeyDown}>
```

Comment:
Fix 

<img width="665" alt="Screenshot 2023-07-16 at 16 23 09" src="https://github.com/mui/material-ui/assets/3165635/d37249f6-078b-4e74-b634-511e75c03f3e">

Per https://www.w3.org/WAI/ARIA/apg/patterns/menubar/examples/menubar-navigation/, 

<img width="962" alt="Screenshot 2023-07-16 at 16 23 39" src="https://github.com/mui/material-ui/assets/3165635/ff817554-4814-4ee2-87f9-2341c463e8de">

it makes more sense to me to use the disclosure pattern as our menu will likely grow to be more complex than what menubar support, this is also what Stripe and Radix did.